src/coreclr/src/vm/typedesc.cpp

```diff
@@ -387,18 +386,12 @@ BOOL TypeDesc::CanCastTo(TypeHandle toTypeHnd, TypeHandlePairList *pVisited)
             case ELEMENT_TYPE_PTR:
                 fCast = TypeDesc::CanCastParam(dac_cast<PTR_ParamTypeDesc>(this)->GetTypeParam(), dac_cast<PTR_ParamTypeDesc>(toTypeDesc)->GetTypeParam(), pVisited);
                 break;
-            case ELEMENT_TYPE_VAR:
-            case ELEMENT_TYPE_MVAR:
-            case ELEMENT_TYPE_FNPTR:
-                fCast = FALSE;
-                break;
             default:
-                BAD_FORMAT_NOTHROW_ASSERT(toKind == ELEMENT_TYPE_TYPEDBYREF || CorTypeInfo::IsPrimitiveType(toKind));
                 // array cast should have been handled above
                 _ASSERTE(toKind != ELEMENT_TYPE_ARRAY);
```

Comment:
These asserts and comment could be removed too. They do not make much sense now. Arrays are no longer handled in this method.